src/docking_verification/src/docking_verification.py

```diff
@@ -1,66 +1,65 @@
-        pub.publish(round(np.mean(moving_average), 4))
-        rate.sleep()
+class DockingVerification():
+    """Check state, publish docking offset from pod."""
+    def __init__(self, average_len=10):
+        self.average_len = average_len
+        self.moving_average = np.zeros(average_len)
+        self.counter = 0
+        self.docking_state = False
 
-def velodyne_points_callback(point_cloud):
-    global counter
-    x = []
-    y = []
-    z = []
-    points = np.zeros(3)
-    for p in pc2.read_points(point_cloud, field_names = ("x", "y", "z"), skip_nans=True):
-        # print (" x : %f  y: %f  z: %f" %(p[0],p[1],p[2]))
-        if abs(p[0]) < 15 and abs(p[1]) < 15 and abs(p[2]) < 0.05:
-            # points = np.append(points, np.array([p[0], p[1], p[2]]), axis = 0)
-            x.append(p[0])
-            y.append(p[1])
-            z.append(p[2])
+        self.publisher = rospy.Publisher('dock_offset', Float64, queue_size=10)
+        self.lidar_sub = rospy.Subscriber('system_status', StateOut,
+                                          self.state_listener)
+        self.lidar_sub = rospy.Subscriber("velodyne_points", PointCloud2,
+                                          self.velodyne_points_callback)
 
-    points = np.zeros((3, len(x)))
-    points[0,:] = np.array(x)
-    points[1,:] = np.array(y)
-    points[1,:] = np.array(z)
+    def velodyne_points_callback(self, point_cloud):
+        """Estimate and publish pod center relative to lidar."""
+        if self.docking_state is False: # no need to calculate
+            return
 
-    # print(len(points[0]))
+        x = y = z = []
+        for p in pc2.read_points(point_cloud, field_names=("x", "y", "z"), skip_nans=True):
+            if abs(p[0]) < 15 and abs(p[1]) < 15 and abs(p[2]) < 0.05:
+                x.append(p[0])
+                y.append(p[1])
+                # z.append(p[2])
 
-    mean_x = np.mean(points[0])
-    mean_y = np.mean(points[1])
-    offset = np.sqrt(mean_x**2 + mean_y**2)
+        points = np.zeros((3, len(x)))
+        points[0, :] = np.array(x)
+        points[1, :] = np.array(y)
+        # points[1, :] = np.array(z)
 
-    moving_average[counter%5] = offset
-    counter += 1
-    # print("Offset ", mean_x, mean_y)
-    # moving_average.append(offset)
-    # print("MA = ", round(sum(moving_average) / len(moving_average), 3 ) )
-    # print("MA = ", round(np.mean(moving_average), 4))
+        mean_x = np.mean(points[0])
+        mean_y = np.mean(points[1])
+        offset = np.sqrt(mean_x**2 + mean_y**2)
 
-def listener():
+        self.moving_average[self.counter % self.average_len] = offset
+        self.counter += 1
 
-    return None
+        #lidar is assumed to be at chassis center
+        self.publisher.publish(round(np.mean(self.moving_average), 4))
 
-if __name__ == '__main__':
+    def state_listener(self, state):
+        """Listen to chassis state and update own state."""
+        if state.DestState == 4:    #TODO: Add in more states?
```

Comment:
Check with Uma, I think she is adding a constant for this state,
It can be used as if state.DestState == state.STATE_'STATE_NAME':